add simplified mesh that will be used if available to find intersection

mockersf opened the pull request:
I'm using `bevy_mod_raycast` to cast rays over generated meshes with many vertices and notice that it sometimes takes a little longer than I would want. As I'm generating the meshes, I'm able to generate lower resolution of the same mesh, and would like to use those for raycasting.

This PR adds a component `SimplifiedMesh` that is optional, and will be used instead of the displayed mesh if available.

From metrics retrieved from tracing:

metric|over complete mesh (40000 vertices) |over simplified mesh (2500 vertices)
-|-|-
avg	 |3.793 ms|       0.290 ms
count	|3,556|    2,693
max	  |8.618 ms|      1.114 ms
min	  |0.009 ms|      0.010 ms
std	    |1.257 ms|    0.098 ms
pct_050	 |4.023 ms|   0.330 ms
pct_090	 |5.170 ms|   0.403 ms

I'm not overly happy with the name of the component, happy to change it to something else

aevyrie commented:
> From metrics retrieved from tracing:

This is a bit OT, but I'm curious if you have any thoughts on benchmarking. I generally use chrome tracing to get some stats, but it still feels unsatisfying. Not directly relevant to this PR, but it would be nice to have a benchmark in place to start tracking perf for significant PRs.

Anyway, I'll take a look at this shortly, this looks like a really awesome addition. Thanks!

aevyrie reviewed (changes requested):
This looks great. My only request is that you add a minimal example in the examples dir. Maybe something as simple as having a high detail and low detail icosphere?

mockersf commented:
> This is a bit OT, but I'm curious if you have any thoughts on benchmarking.

It would require a bit to extract the main logic so that it's less dependant on Bevy, but it should be possible

> My only request is that you add a minimal example in the examples dir

Added an example where you can toggle the `SimplifiedMesh` with the spacebar. I switched to a uvsphere instead of a icosphere as the mesh can be more complicated, and used values that makes it sluggish even in release mode on my (not very good) laptop
